frontend/src/lib/components/widgets/NumberInput/NumberInput.tsx

```diff
@@ -273,12 +311,14 @@ class NumberInput extends React.PureComponent<Props, State> {
 
   /** True if the input's current value can be decremented by its step. */
   private get canDecrement(): boolean {
-    return this.state.value - this.getStep() >= this.getMin()
+    const value = this.state.value ? this.state.value : 0
```

Comment:
what's the `value` type? Is it:

```
const value: int
```
?

If yes, then why do we check something about it on lines 336 and  349?